lms/static/scripts/frontend_apps/components/test/StudentSelector-test.js

```diff
@@ -0,0 +1,96 @@
+import { createElement } from 'preact';
+import { shallow } from 'enzyme';
+
+import StudentSelector from '../StudentSelector';
+
+describe('StudentSelector', () => {
+  const renderSelector = (props = {}) => {
+    const fakeOnSelectStudent = sinon.stub();
+    const fakeSelectedStudentIndex = 0;
+    const fakeStudents = [
+      {
+        username: 'user1',
+        displayName: 'User 1',
+      },
+      {
+        username: 'user2',
+        displayName: 'User 2',
+      },
+    ];
+
+    return shallow(
+      <StudentSelector
+        onSelectStudent={fakeOnSelectStudent}
+        selectedStudentIndex={fakeSelectedStudentIndex}
+        students={fakeStudents}
+        {...props}
+      />
+    );
+  };
+
+  it('shall not have an initial selected option', () => {
+    const wrapper = renderSelector({ selectedStudentIndex: -1 });
+    assert.equal(
+      wrapper.find('select [selected=true]').text(),
+      'Select a student'
+    );
+  });
+
+  it('sets the selected option should to the second user', () => {
+    const wrapper = renderSelector({ selectedStudentIndex: 1 });
+    assert.equal(wrapper.find('select [selected=true]').text(), 'User 2');
+  });
+
+  it('calls the onSelectStudent callback when the select list is changed', () => {
+    const onChange = sinon.spy();
+    const wrapper = renderSelector({ onSelectStudent: onChange });
+    wrapper.find('select').simulate('change');
+    assert.isTrue(onChange.called);
+  });
+
+  it('calls the callback with the next student index', () => {
```

Comment:
Should this test name reflect that there is a click involved here? And/or which button it's clicking on?

Reply:
Its not the click so much, but rather testing that the onSelectStudent function does what its supposed to do. I'll reword this